### PR TITLE
[AMD] Add AMD ROCm support for HiCache JIT kernel benchmark

### PR DESCRIPTION
**Pull request by Emmanuel0612**
## Motivation

To complete the CI benchmark of JIT kernels with AMD devices.

## Modifications

We had added register_amd_ci in the following file.
sglang/python/sglang/jit_kernel/benchmark/bench_hicache.py

## Accuracy Tests
image

## Speed Tests and Profiling

root@smci355-ccs-aus-n10-17:/sgl-workspace/sglang/test# python run_suite.py --hw amd --suite stage-b-test-1-gpu-large-amd
+------------+------------------------------+-----------+-------------+
| Hardware | Suite | Nightly | Partition |
|------------+------------------------------+-----------+-------------|
| AMD | stage-b-test-1-gpu-large-amd | False | full |
+------------+------------------------------+-----------+-------------+

python3 /sgl-workspace/sglang/python/sglang/jit_kernel/benchmark/bench_hicache.py

One Layer: Host -> Device (CPU -> GPU)

hicache-one-layer-h2d:
    element_size  batch_size  SGL AOT Kernel (us)  SGL JIT Kernel (us)  PyTorch (us)
0           64.0         1.0             4.405125             6.417562     45.675501
1           64.0         2.0             4.915000             6.275000     44.842875
2           64.0         4.0             6.370125             6.235000     44.940501
3           64.0         8.0             7.485000             7.125000     48.487937
4           64.0        16.0            10.270125             7.067500     49.330501
5           64.0        32.0            10.825125             8.887625     50.750501
6           64.0        64.0            18.750126            14.250125     85.058376
7           64.0       128.0            34.105375            28.002750     88.090748
8           64.0       256.0            62.925503            21.505187    100.190878
9           64.0       512.0           121.481128            30.825250    102.045938
10          64.0      1024.0           213.244498            51.565502    107.051000
11          64.0      2048.0           371.118501            90.945877    118.051127
12          64.0      4096.0           668.331265           158.933997    149.888873
13          64.0      8192.0          1208.851218           293.547750    211.484432
14          64.0     16384.0          2312.746525           450.736612    179.654121
15          64.0     32768.0          4611.318111           731.504232    284.530118
16         128.0         1.0             5.320125             7.250125     49.087938
17         128.0         2.0             5.590000             7.242562     48.232999
18         128.0         4.0             6.350000             7.230063     48.765499
19         128.0         8.0             7.110125             7.200000     49.165498
20         128.0        16.0             9.410125             7.120063     49.948001
21         128.0        32.0            10.050063             9.145125     86.850747
22         128.0        64.0            17.075188            16.045125     89.065872
23         128.0       128.0            31.392813            30.595249    104.933500
24         128.0       256.0            60.303001            31.617751    106.505997
25         128.0       512.0           116.586126            52.185498    111.915998
26         128.0      1024.0           213.667005            90.770751    121.548623
27         128.0      2048.0           390.248626           155.936375    156.661376
28         128.0      4096.0           712.021649           299.247816    258.547366
29         128.0      8192.0          1281.254470           569.050252    191.411749
30         128.0     16384.0          2317.386389          1050.562322    292.152762
31         128.0     32768.0          4289.615154          1770.246506    508.684814
32         256.0         1.0             5.510000             7.175000     48.765499
33         256.0         2.0             5.790062             7.285125     48.338002
34         256.0         4.0             6.480125             7.335000     49.155377
35         256.0         8.0             7.855000             7.205125     49.965374
36         256.0        16.0             9.965125             7.140000     86.940750
37         256.0        32.0            10.162563             9.670000     88.515878
38         256.0        64.0            17.487625            14.965188    105.290998
39         256.0       128.0            32.645313            33.485249    106.695939
40         256.0       256.0            64.088129            36.140375    111.333564
41         256.0       512.0           124.911189            75.215749    120.471124
42         256.0      1024.0           232.042186           134.796247    149.391368
43         256.0      2048.0           431.113988           231.552124    237.152182
44         256.0      4096.0           802.994937           431.544006    195.736870
45         256.0      8192.0          1452.673554           788.907349    290.532738
46         256.0     16384.0          2627.784491          1520.079136    618.738294
47         256.0     32768.0          4739.853859          2908.082008   1383.602858
48         512.0         1.0             6.557563             7.282563     49.315501
49         512.0         2.0             8.050125             7.245125     48.410499
50         512.0         4.0             8.615000             7.327562     49.348000
51         512.0         8.0             9.907562             7.660125     87.070875
52         512.0        16.0            12.440125             8.290125     87.248314
53         512.0        32.0            12.465125            13.772625    104.341000
54         512.0        64.0            21.815250            24.225250    106.888499
55         512.0       128.0            40.860374            30.495251    111.271001
56         512.0       256.0            79.895750            65.495498    122.528624
57         512.0       512.0           154.486373           137.738749    152.043939
58         512.0      1024.0           296.262741           258.687377    249.867380
59         512.0      2048.0           578.722894           463.819385    190.289311
60         512.0      4096.0          1114.460409           877.723277    529.514879
61         512.0      8192.0          2167.930365          1715.255976    488.454565
62         512.0     16384.0          4121.288300          3409.266949   1177.228570
63         512.0     32768.0          7610.401154          6505.755901   3979.902029
64        1024.0         1.0            10.805000             7.217563     49.105437
65        1024.0         2.0            12.140125             7.215000     49.165498
66        1024.0         4.0            12.855000             7.307625     87.953378
67        1024.0         8.0            14.065125             8.585125     88.870876
68        1024.0        16.0            16.520126            10.020125     94.210878
69        1024.0        32.0            16.420124            17.395126     95.825877
70        1024.0        64.0            30.705250            17.845126     99.425875
71        1024.0       128.0            58.530625            35.687812    110.091001
72        1024.0       256.0           114.276126            78.320749    135.841250
73        1024.0       512.0           224.337064           153.931379    222.602129
74        1024.0      1024.0           438.699067           319.617867    174.846619
75        1024.0      2048.0           856.972992           664.718747    269.294932
76        1024.0      4096.0          1679.475665          1328.462362    443.126634
77        1024.0      8192.0          3366.071463          2694.925070    960.483938
78        1024.0     16384.0          6617.197037          5333.304882   3466.597557
79        1024.0     32768.0         12704.168320         10179.334641   6140.857220

All Layer: Device -> Host (GPU -> CPU) [per-layer avg]

hicache-all-layer-d2h:
    element_size  batch_size  SGL AOT Kernel (us)  SGL JIT Kernel (us)  PyTorch (us)
0           64.0         1.0             1.470000             1.285000     48.445376
1           64.0         2.0             2.975125             2.605000     48.545375
2           64.0         4.0             3.730000             2.510000     48.200501
3           64.0         8.0             5.435063             3.575125     48.415501
4           64.0        16.0            10.177562             3.685125     48.898000
5           64.0        32.0            10.335125             6.100125     50.080501
6           64.0        64.0            19.790124            11.320125     80.543250
7           64.0       128.0            38.225375            23.695249     84.935747
8           64.0       256.0            74.128129            20.655250    103.798497
9           64.0       512.0           141.623877            36.110375    109.516002
10          64.0      1024.0           262.397498            65.165497    120.326124
11          64.0      2048.0           519.359887           126.281127    141.496256
12          64.0      4096.0          1030.367076           246.337369    188.801751
13          64.0      8192.0          2044.249058           452.659115    245.649807
14          64.0     16384.0          4068.092823           834.462881    202.049375
15          64.0     32768.0          8022.080421          1493.203819    324.628055
16         128.0         1.0             2.570000             2.375000     48.620438
17         128.0         2.0             2.825000             2.560000     48.505500
18         128.0         4.0             3.490125             2.630000     48.427938
19         128.0         8.0             5.465063             2.815000     48.687937
20         128.0        16.0            10.045125             3.830000     49.925499
21         128.0        32.0             9.752625             7.277563     79.910688
22         128.0        64.0            18.625125            13.602625     85.240752
23         128.0       128.0            36.515374            28.347750    103.661001
24         128.0       256.0            72.850626            31.622814    111.575998
25         128.0       512.0           139.211312            53.055499    122.593626
26         128.0      1024.0           267.897487            94.920874    146.648876
27         128.0      2048.0           528.844893           174.141631    208.389379
28         128.0      4096.0          1015.181959           339.135617    241.302252
29         128.0      8192.0          2050.714016           659.571171    193.366751
30         128.0     16384.0          4017.762661          1265.909314    313.572884
31         128.0     32768.0          8165.081024          2363.737106    578.517854
32         256.0         1.0             2.575000             2.500000     48.620500
33         256.0         2.0             2.960000             2.765125     48.310500
34         256.0         4.0             3.785000             2.920000     49.170500
35         256.0         8.0             5.930125             3.830125     50.245501
36         256.0        16.0            11.085000             4.490125     80.565751
37         256.0        32.0            10.560125             7.432563     85.140750
38         256.0        64.0            20.130251            14.260125    103.325874
39         256.0       128.0            39.150376            34.925375    109.626003
40         256.0       256.0            76.825749            37.130374    119.431123
41         256.0       512.0           150.236368            80.903251    143.633813
42         256.0      1024.0           287.892640           138.138749    184.441693
43         256.0      2048.0           546.552569           244.939752    243.687250
44         256.0      4096.0          1070.564985           455.986753    198.331878
45         256.0      8192.0          2130.744696           852.028012    298.210323
46         256.0     16384.0          4269.945145          1710.291028    548.705101
47         256.0     32768.0          8519.904137          3301.770687   1214.101315
48         512.0         1.0             3.640000             2.730000     48.045499
49         512.0         2.0             3.962563             2.905000     48.810437
50         512.0         4.0             5.440000             3.250000     49.705502
51         512.0         8.0             7.600000             3.620125     80.830749
52         512.0        16.0            14.467625             6.835125     86.393375
53         512.0        32.0            14.180125            11.510062    104.026001
54         512.0        64.0            26.700250            24.762687    108.446002
55         512.0       128.0            52.120499            26.505250    119.231001
56         512.0       256.0           102.871001            61.565626    142.111257
57         512.0       512.0           195.686877           141.206369    183.446690
58         512.0      1024.0           379.816070           254.387379    240.307249
59         512.0      2048.0           743.746877           497.524634    200.286880
60         512.0      4096.0          1439.548492           966.543972    304.380372
61         512.0      8192.0          2922.602177          1931.632996    521.859884
62         512.0     16384.0          5797.974110          3730.839729   1118.725359
63         512.0     32768.0         11712.634087          7153.436661   7356.343269
64        1024.0         1.0             4.990000             2.715000     46.900377
65        1024.0         2.0             5.067562             2.735000     49.570499
66        1024.0         4.0             7.297563             3.770000     79.838250
67        1024.0         8.0             9.420125             5.260000     86.835813
68        1024.0        16.0            17.412688             8.980125    104.248501
69        1024.0        32.0            16.890250            15.387625    110.958561
70        1024.0        64.0            33.275373            17.210126    119.576126
71        1024.0       128.0            63.638125            32.975249    141.766369
72        1024.0       256.0           124.596126            68.680622    183.649182
73        1024.0       512.0           237.514690           148.066372    240.207247
74        1024.0      1024.0           460.754365           330.998003    187.291749
75        1024.0      2048.0           903.885931           711.596608    287.765265
76        1024.0      4096.0          1791.231751          1481.223881    490.574628
77        1024.0      8192.0          3581.913471          3040.848494   1026.314497
78        1024.0     16384.0          7220.396996          6020.495892   7182.267189
79        1024.0     32768.0         14777.418137         11570.262909  13353.874207

filename='/sgl-workspace/sglang/python/sglang/jit_kernel/benchmark/bench_hicache.py', elapsed=61, estimated_time=56.0

✓ PASSED:

/sgl-workspace/sglang/python/sglang/jit_kernel/benchmark/bench_hicache.py

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

---
### CI States

Latest PR Test (Base): :no_entry_sign: [Run #26144024915](https://github.com/sgl-project/sglang/actions/runs/26144024915)
Latest PR Test (Extra): :x: [Run #26144024820](https://github.com/sgl-project/sglang/actions/runs/26144024820)

### CODE REVIEW

File: python/sglang/jit_kernel/benchmark/bench_hicache.py [hunk 1]
```diff
@@ -22,6 +22,10 @@
 
 import torch
 import triton
+
+if not hasattr(triton, "__version__"):
+    triton.__version__ = "custom-amd-build"
+
```
Comment (HaiShaw) [inline python/sglang/jit_kernel/benchmark/bench_hicache.py]: This seems to be problematic.

**Review by HaiShaw — CHANGES_REQUESTED** [review]

**Comment by yctseng0211** [comment]
image
CI passed
